OpenPrFunction/PullRequest.cs

```diff
@@ -29,12 +30,39 @@ public async Task<Pr> OpenAsync(GitHubClientParameters parameters, Settings sett
             }
 
             var stats = Stats.ParseStats(commit.Commit.Message);
-            var pr = new NewPullRequest(KnownGitHubs.CommitMessageTitle, KnownGitHubs.BranchName, baseBranch)
-            {
-                Body = PullRequestBody.Generate(stats),
-            };
 
-            var result = await githubClient.PullRequest.Create(parameters.RepoOwner, parameters.RepoName, pr);
+            Octokit.PullRequest result;
+            if(update) {
+                //get PR number
+                var allPrs = await githubClient.PullRequest.GetAllForRepository(parameters.RepoOwner, parameters.RepoName);
+                
+                var prNumber = 0;
+                foreach(var pr in allPrs) {
```

Comment:
Use Linq syntax to search in list
```
var pr = allPrs.First( pr => pr.State == ItemState.Open && pr.Head.Sha == branch.Commit.Sha);

if (pr == null)  { //throw}
```

Reply:
I tried to implement it like this but wasn't able to make it work ....